GPU/Common/VertexShaderGenerator.cpp

```diff
@@ -1052,6 +1052,9 @@ bool GenerateVertexShader(const VShaderID &id, char *buffer, const ShaderLanguag
 					WRITE(p, "  %sv_color0 = color0;\n", compat.vsOutPrefix);
 			} else {
 				WRITE(p, "  %sv_color0 = u_matambientalpha;\n", compat.vsOutPrefix);
+				if (bugs.Has(Draw::Bugs::MALI_CONSTANT_LOAD_BUG) && g_Config.bVendorBugChecksEnabled) {
+					WRITE(p, "  %sv_color0.r += 0.000001;\n", compat.vsOutPrefix);
```

Comment:
Just wondering how good is the driver to optimize cancelling out operation?

Like this:
```
WRITE(p, "  %sv_color0.r += 0.000001;\n", compat.vsOutPrefix);
WRITE(p, "  %sv_color0.r -= 0.000001;\n", compat.vsOutPrefix);
```

Or even better to avoid float precision problems:
```
WRITE(p, "  %sv_color0.xy = %sv_color0.yx;\n", compat.vsOutPrefix, compat.vsOutPrefix);
WRITE(p, "  %sv_color0.xy = %sv_color0.yx;\n", compat.vsOutPrefix, compat.vsOutPrefix);
```

Reply:
I tried the first one, it optimized it out. Yet to try the second.